Packs/1Password/Integrations/OnePasswordEventCollector/OnePasswordEventCollector.py

```diff
@@ -0,0 +1,392 @@
+    return arg_to_number(params.get(param_name)) or DEFAULT_MAX_EVENTS_PER_FETCH
+
+
+def get_unauthorized_event_types(auth_introspection_response: dict[str, Any], event_types: list[str]) -> list[str]:
+    """Checks if the bearer token has no access to any of the configured event types.
+
+    Args:
+        auth_introspection_response (dict): Response JSON from Client.introspect call.
+        event_types (list): List of event types from the integration configuration params.
+
+    Returns:
+        list[str]: List of unauthorized event types that the token does not have access to.
+    """
+    authorized_features = auth_introspection_response['features']
+    return [
+        event_type for event_type in event_types
+        if EVENT_TYPE_MAPPING.get(event_type) not in authorized_features
+    ]
+
+
+def add_fields_to_event(event: dict[str, Any], event_type: str):
+    """Sets the '_time' and 'event_type' fields to an event.
+
+    Args:
+        event (dict): Event dictionary with the new fields.
+        event_type (str): Type of 1Password event (e.g. 'Item usage actions', 'Sign in attempts').
+    """
+    event_time = arg_to_datetime(event['timestamp'], required=True)
+    event['_time'] = event_time.strftime(DATE_FORMAT)  # type: ignore[union-attr]
+    event['event_type'] = event_type
+
+
+def get_events_from_client(
+    client: Client,
+    event_type: str,
+    from_date: datetime,
+    max_events: int,
+    ids_to_skip: set[str] | None = None
+) -> list[dict]:
+    """Gets events of the specified type based on the 'from_date' filter and 'max_events' argument.
+
+    Args:
+        client (Client): 1Password Events API client.
+        event_type (str): Type of 1Password event (e.g. 'Item usage actions', 'Sign in attempts').
+        from_date (datetime): Datetime from which to get events.
+        max_events (int): The maximum number of events to fetch.
+        ids_to_skip (set | None): Optional set of already-fetched event UUIDs that should be skipped.
+
+    Returns:
+        list[dict]: List of events.
+    """
+    events: list[dict] = []
+    ids_to_skip = ids_to_skip or set()
+
+    # First call to the paginated API needs to to include a date filter
+    has_more_events = True
+    client_kwargs: dict[str, Any] = {'event_type': event_type, 'from_date': from_date}
+
+    while has_more_events:
+        response = client.get_events(**client_kwargs)
+        pagination_cursor = response['cursor']
+
+        for event in response['items']:
+            event_id = event['uuid']
+
+            if event_id in ids_to_skip:
+                continue
+
+            if len(events) == max_events:
+                break
+
+            add_fields_to_event(event, event_type)
+            events.append(event)
+            ids_to_skip.add(event_id)
+
+        demisto.debug(
+            f'Response has {len(response["items"])} events and pagination cursor: {pagination_cursor}.'
```

Comment:
add the number of events after filtering out ignored ones

Reply:
Added